android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt

```diff
@@ -50,1235 +50,773 @@ import com.bitmovin.player.api.ui.UiConfig
-         * @return The generated `TimelineReferencePoint`.
-         */
-        @JvmStatic
-        private fun toTimelineReferencePoint(json: String?): TimelineReferencePoint? = when (json) {
-            "start" -> TimelineReferencePoint.Start
-            "end" -> TimelineReferencePoint.End
-            else -> null
-        }
+/**
+ * Converts an arbitrary `json` to `AdaptationConfig`.
+ */
+private fun ReadableMap.toAdaptationConfig(): AdaptationConfig = AdaptationConfig().apply {
+    withInt("maxSelectableBitrate") { maxSelectableVideoBitrate = it }
+}
 
-        /**
-         * Converts an arbitrary `json` to `AdaptationConfig`.
-         * @param json JS object representing the `AdaptationConfig`.
-         * @return The generated `AdaptationConfig` if successful, `null` otherwise.
-         */
-        private fun toAdaptationConfig(json: ReadableMap?): AdaptationConfig? {
-            if (json == null) return null
-            val adaptationConfig = AdaptationConfig()
-            if (json.hasKey("maxSelectableBitrate")) {
-                adaptationConfig.maxSelectableVideoBitrate = json.getInt("maxSelectableBitrate")
-            }
-            return adaptationConfig
-        }
+/**
+ * Converts any JS object into a `PlaybackConfig` object.
+ */
+fun ReadableMap.toPlaybackConfig(): PlaybackConfig = PlaybackConfig().apply {
+    withBoolean("isAutoplayEnabled") { isAutoplayEnabled = it }
+    withBoolean("isMuted") { isMuted = it }
+    withBoolean("isTimeShiftEnabled") { isTimeShiftEnabled = it }
+}
 
-        /**
-         * Converts any JS object into a `PlaybackConfig` object.
-         * @param json JS object representing the `PlaybackConfig`.
-         * @return The generated `PlaybackConfig` if successful, `null` otherwise.
-         */
-        @JvmStatic
-        fun toPlaybackConfig(json: ReadableMap?): PlaybackConfig? {
-            if (json == null) {
-                return null
-            }
-            val playbackConfig = PlaybackConfig()
-            if (json.hasKey("isAutoplayEnabled")) {
-                playbackConfig.isAutoplayEnabled = json.getBoolean("isAutoplayEnabled")
-            }
-            if (json.hasKey("isMuted")) {
-                playbackConfig.isMuted = json.getBoolean("isMuted")
-            }
-            if (json.hasKey("isTimeShiftEnabled")) {
-                playbackConfig.isTimeShiftEnabled = json.getBoolean("isTimeShiftEnabled")
-            }
-            return playbackConfig
-        }
+/**
+ * Converts any JS object into a `StyleConfig` object.
+ */
+fun ReadableMap.toStyleConfig(): StyleConfig = StyleConfig().apply {
+    withBoolean("isUiEnabled") { isUiEnabled = it }
+    getString("playerUiCss")?.takeIf { it.isNotEmpty() }?.let { playerUiCss = it }
+    getString("supplementalPlayerUiCss")?.takeIf { it.isNotEmpty() }?.let { supplementalPlayerUiCss = it }
+    getString("playerUiJs")?.takeIf { it.isNotEmpty() }?.let { playerUiJs = it }
```

Comment:
is the filtering on empty needed?

Reply:
I left it as is as this refactor doesn't intend to introduce any behavior differences.